LandSandBoat/server
# Add mobMod ENCROACH_TARGET for NMs like Claret and Dyinyinga

**_I affirm:_**

## What does this pull request do?


This PR adds a new mob mod, ENCROACH_TARGET, which when enabled on a mob allows the mob to "close in" on its target closer than is normally allowed.  This functionality is used by NMs like Claret and Dyinyinga where they apply a status to the player if they are able to get close enough.  These NMs attempt to make model to model contact with players.

## Steps to test these changes

1. Get to Claret  
> !zone {Mount Zhayolm}   
> !pos 501 -9 53)
2. Turn off GodMode if on.  Enable !Immortal if not on
3. Get a pectin
>!additem 2591 
4. !hp 0 the nearby clot or toggle GM
5. Trade the pectin to the ??? to spawn Claret
6. Note that Claret will get very close to the player, much closer than normal mobs, and a poison is applied.
7. Move away from Claret slowly, note that Claret continues to close in tightly as soon as the player is a little bit away
8. Run from Claret (with speed boost/flee), note that when Claret finally catches up, it encroaches on the player
9. Aggro/pull a clot nearby, note that the clot does not encroach the player even though Claret still does.
